### PR TITLE
HDDS-1691 : RDBTable#isExist should use Rocksdb#keyMayExist

### PR DESCRIPTION
**Pull request by avijayanhwx**
RDBTable#isExist can use Rocksdb#keyMayExist, this avoids the cost of reading the value for the key.

Please refer, 
https://github.com/facebook/rocksdb/blob/7a8d7358bb40b13a06c2c6adc62e80295d89ed05/java/src/main/java/org/rocksdb/RocksDB.java#L2184

### CODE REVIEW

File: hadoop-hdds/common/src/test/java/org/apache/hadoop/utils/db/TestRDBStore.java [hunk 1]
```diff
@@ -289,4 +290,41 @@ public void testRocksDBCheckpointCleanup() throws Exception {
...
+            StringUtils.getBytesUtf16("key" + i))!= null);
+      }
+      long end = System.nanoTime();
+      long keyGetLatency = end - start;
+
+      start = System.nanoTime();
+      for (int i = 50; i < 100; i++) {
+        Assert.assertTrue(db.get(
+            StringUtils.getBytesUtf16("key" + i))!= null);
+      }
```
Comment (bharatviswa504) [inline hadoop-hdds/common/src/test/java/org/apache/hadoop/utils/db/TestRDBStore.java]: I have not understood how this test is actually testing the API?

Comment (avijayanhwx) [reply hadoop-hdds/common/src/test/java/org/apache/hadoop/utils/db/TestRDBStore.java]: Updated the UT. Hopefully this is clearer

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 33 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 536 | trunk passed |
| +1 | compile | 257 | trunk passed |
| +1 | checkstyle | 63 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 803 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 170 | trunk passed |
| 0 | spotbugs | 301 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 493 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 450 | the patch passed |
| +1 | compile | 268 | the patch passed |
| +1 | javac | 268 | the patch passed |
| +1 | checkstyle | 75 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 672 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 158 | the patch passed |
| +1 | findbugs | 548 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 268 | hadoop-hdds in the patch failed. |
| -1 | unit | 1445 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 6443 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestSCMContainerPlacementRackAware |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1013 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux a79c520ec659 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 049bb42 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/1/testReport/ |
| Max. process+thread count | 5069 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common U: hadoop-hdds/common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for branch |
| +1 | mvninstall | 474 | trunk passed |
| +1 | compile | 243 | trunk passed |
| +1 | checkstyle | 60 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 767 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 148 | trunk passed |
| 0 | spotbugs | 299 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 483 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 28 | Maven dependency ordering for patch |
| +1 | mvninstall | 416 | the patch passed |
| +1 | compile | 249 | the patch passed |
| +1 | javac | 249 | the patch passed |
| +1 | checkstyle | 69 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 603 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 140 | the patch passed |
| +1 | findbugs | 512 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 89 | hadoop-hdds in the patch failed. |
| -1 | unit | 1030 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 54 | The patch does not generate ASF License warnings. |
| | | 5566 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.utils.db.TestRDBStore |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1013 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 62edc012c76d 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 049bb42 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/4/testReport/ |
| Max. process+thread count | 4967 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/ozone-recon U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 34 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 38 | Maven dependency ordering for branch |
| +1 | mvninstall | 552 | trunk passed |
| +1 | compile | 257 | trunk passed |
| +1 | checkstyle | 62 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 846 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 165 | trunk passed |
| 0 | spotbugs | 366 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 586 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for patch |
| -1 | mvninstall | 32 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 81 | hadoop-ozone in the patch failed. |
| -1 | compile | 95 | hadoop-hdds in the patch failed. |
| -1 | javac | 95 | hadoop-hdds in the patch failed. |
| +1 | checkstyle | 69 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | shadedclient | 653 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 164 | the patch passed |
| +1 | findbugs | 516 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 231 | hadoop-hdds in the patch passed. |
| -1 | unit | 1324 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 49 | The patch does not generate ASF License warnings. |
| | | 6103 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1013 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 757f649710f3 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 049bb42 |
| Default Java | 1.8.0_212 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/artifact/out/patch-compile-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/testReport/ |
| Max. process+thread count | 5007 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/ozone-recon U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 29 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 65 | Maven dependency ordering for branch |
| +1 | mvninstall | 471 | trunk passed |
| +1 | compile | 261 | trunk passed |
| +1 | checkstyle | 73 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 860 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 161 | trunk passed |
| 0 | spotbugs | 313 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 501 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 36 | Maven dependency ordering for patch |
| +1 | mvninstall | 433 | the patch passed |
| +1 | compile | 268 | the patch passed |
| +1 | javac | 268 | the patch passed |
| +1 | checkstyle | 71 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 670 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 162 | the patch passed |
| +1 | findbugs | 512 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 250 | hadoop-hdds in the patch passed. |
| -1 | unit | 1164 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 6190 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1013 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 7bc75164bc37 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 049bb42 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/3/testReport/ |
| Max. process+thread count | 3824 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/ozone-recon U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 31 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for branch |
| +1 | mvninstall | 514 | trunk passed |
| +1 | compile | 246 | trunk passed |
| +1 | checkstyle | 63 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 803 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 153 | trunk passed |
| 0 | spotbugs | 302 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 489 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 29 | Maven dependency ordering for patch |
| +1 | mvninstall | 416 | the patch passed |
| +1 | compile | 258 | the patch passed |
| +1 | javac | 258 | the patch passed |
| +1 | checkstyle | 68 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 616 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 148 | the patch passed |
| +1 | findbugs | 504 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 258 | hadoop-hdds in the patch passed. |
| -1 | unit | 1640 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 41 | The patch does not generate ASF License warnings. |
| | | 6445 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.safemode.TestSCMSafeModeWithPipelineRules |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1013 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux aa736653db1a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 049bb42 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/5/testReport/ |
| Max. process+thread count | 4862 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/ozone-recon U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by avijayanhwx** [comment]
Test failures are unrelated.
`
org.apache.hadoop.ozone.client.rpc.TestBCSID.org.apache.hadoop.ozone.client.rpc.TestBCSID
org.apache.hadoop.ozone.client.rpc.TestFailureHandlingByClient.testContainerExclusionWithClosedContainerException
`

**Comment by avijayanhwx** [comment]
cc @bharatviswa504 , @mukul1987, @nandakumar131

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/RDBTable.java [hunk 2]
```diff
@@ -122,7 +122,11 @@ public boolean isEmpty() throws IOException {
   @Override
   public boolean isExist(byte[] key) throws IOException {
     try {
-      return db.get(handle, key) != null;
+      // RocksDB#keyMayExist
+      // If the key definitely does not exist in the database, then this
+      // method returns false, else true.
+      return db.keyMayExist(handle, key, new StringBuilder())
```
Comment (bharatviswa504) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/RDBTable.java]: As discussed offline, can we add some tests with proto objects and see will it cause any issues.

As our data in DB is byteArray of proto-objects.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 30 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 3 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 20 | Maven dependency ordering for branch |
| +1 | mvninstall | 475 | trunk passed |
| +1 | compile | 249 | trunk passed |
| +1 | checkstyle | 62 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 786 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 145 | trunk passed |
| 0 | spotbugs | 308 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 501 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 432 | the patch passed |
| +1 | compile | 241 | the patch passed |
| +1 | javac | 241 | the patch passed |
| +1 | checkstyle | 63 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | shadedclient | 624 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 142 | the patch passed |
| +1 | findbugs | 529 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 264 | hadoop-hdds in the patch failed. |
| -1 | unit | 1254 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 39 | The patch does not generate ASF License warnings. |
| | | 6033 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestSCMContainerPlacementRackAware |
|   | hadoop.ozone.om.TestOzoneManager |
|   | hadoop.ozone.ozShell.TestOzoneShell |
|   | hadoop.ozone.scm.node.TestQueryNode |
|   | hadoop.ozone.scm.TestAllocateContainer |
|   | hadoop.ozone.ozShell.TestS3Shell |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1013 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux c024b67e3705 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a79bdf7 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/7/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/7/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/7/testReport/ |
| Max. process+thread count | 1919 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/integration-test hadoop-ozone/ozone-recon U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 3 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 63 | Maven dependency ordering for branch |
| +1 | mvninstall | 469 | trunk passed |
| +1 | compile | 245 | trunk passed |
| +1 | checkstyle | 63 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 775 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 143 | trunk passed |
| 0 | spotbugs | 319 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 504 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 28 | Maven dependency ordering for patch |
| +1 | mvninstall | 431 | the patch passed |
| +1 | compile | 246 | the patch passed |
| +1 | javac | 246 | the patch passed |
| -0 | checkstyle | 33 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 619 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 142 | the patch passed |
| +1 | findbugs | 529 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 259 | hadoop-hdds in the patch failed. |
| -1 | unit | 1654 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 51 | The patch does not generate ASF License warnings. |
| | | 6468 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestSCMContainerPlacementRackAware |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1013 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 8a62fdeafa9a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a79bdf7 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/6/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/6/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/6/testReport/ |
| Max. process+thread count | 4931 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/integration-test hadoop-ozone/ozone-recon U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1013/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**